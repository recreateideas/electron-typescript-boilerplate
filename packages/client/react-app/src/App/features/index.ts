export * from './AuthenticatedRoute';
export * from './ThemeProviderConnected';
