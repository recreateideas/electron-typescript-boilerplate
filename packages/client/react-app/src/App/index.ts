export * from './ServicedApp';
export * from './App';
