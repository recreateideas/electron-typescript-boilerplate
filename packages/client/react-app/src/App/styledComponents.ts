import styled from 'styled-components';

export const Application = styled.div``;
