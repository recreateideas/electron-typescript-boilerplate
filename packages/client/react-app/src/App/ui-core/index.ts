export * from './theme';
export * from './Loader';
