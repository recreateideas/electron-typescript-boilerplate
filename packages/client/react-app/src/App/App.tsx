import { ReactElement } from 'react';
import { ConnectedRouter as Router } from 'connected-react-router';
import { Switch, Redirect, Route } from 'react-router-dom';
import { history } from '../redux';
import { routes } from '../routes';
import { AuthenticatedRoute } from './features';
import { Application } from './styledComponents';

export const App = (): ReactElement => {
    return (
        <Application>
            <Router history={history}>
                {/* <AuthenticatedRoute render={(props) => <SideBar {...props} />} /> */}
                <Switch>
                    {Object.values(routes).map((routeObj, i) => {
                        const {
                            title,
                            exact,
                            path,
                            components: { mainView: Component },
                        } = routeObj;
                        return (
                            <AuthenticatedRoute
                                key={i}
                                exact={exact}
                                path={path}
                                title={title}
                                render={(props) => (
                                    // <ErrorBoundary>
                                    //     <PageContainer>
                                    //         <Header title={title} />
                                    //         <PageContent className="page-content">
                                    <Component {...props} />
                                    //         </PageContent>
                                    //     </PageContainer>
                                    // </ErrorBoundary>
                                )}
                            />
                        );
                    })}
                    <Route path="/">
                        <Redirect exact to="/editor" />
                    </Route>
                </Switch>
            </Router>
        </Application>
    );
};

App.displayName = 'App';

App.defaultProps = {};
