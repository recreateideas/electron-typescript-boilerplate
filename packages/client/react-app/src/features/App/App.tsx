import { useHealthCheck, useServicePorts } from '../../hooks';
import { Loader } from '../../ui-core';

function App() {
    const servicePorts = useServicePorts();
    const isHealthy = useHealthCheck({ servicePorts });
    const isChecking = isHealthy === undefined;
    return (
        <div id="app">
            {isHealthy ? (
                <div>Healthy</div>
            ) : !servicePorts ? (
                <Loader
                    {...{
                        className: 'setting-up',
                        color: 'green',
                        message: 'setting up services...',
                    }}
                />
            ) : isChecking ? (
                <Loader
                    {...{
                        className: 'checking',
                        color: '#6b5b95',
                        message: 'checking services...',
                    }}
                />
            ) : (
                <Loader
                    {...{
                        className: 'not-responding',
                        color: 'red',
                        freeze: true,
                        message: 'Services are not responding. Cmd + R to retry.',
                    }}
                />
            )}
        </div>
    );
}

export { App };
