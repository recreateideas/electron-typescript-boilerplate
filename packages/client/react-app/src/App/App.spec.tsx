import { App } from './App';
import { TestProvider } from '../tests';

const { mountWithRouterProps } = TestProvider();
describe('App', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithRouterProps(<App />, state);
        expect(root).toMatchSnapshot();
    });
});
