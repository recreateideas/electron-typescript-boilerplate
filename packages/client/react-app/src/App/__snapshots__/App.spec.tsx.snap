// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="sc-bdnxRM"
>
  <div>
    Editor
  </div>
</div>
`;
